gammapy/maps/geom.py

```diff
@@ -1962,6 +1977,17 @@ def data_shape(self):
         """Shape of the Numpy data array matching this geometry."""
         pass
 
+    def data_nbytes(self, dtype="float32"):
+        """Estimate memory usage in bytes of the Numpy data array
+        matching this geometry depending on the given type.
+
+        Parameters
+        ----------
+        dtype : data-type
+            The desired data-type for the array. Default is "float32"
+        """
+        return np.empty(self.data_shape, dtype).nbytes
```

Comment:
Maybe return a `Quantity` object here and convert to `MB` by default? And add the `Returns` section to the docstring...